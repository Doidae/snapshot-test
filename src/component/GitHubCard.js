import React from 'react';
import { Card } from 'react-bootstrap';

const GitHubCard = ({ username, imageUrl, blurb }) => {
    return (
        <Card style={{ width: '18rem' }}>
            <Card.Img variant="top" src={imageUrl} />
            <Card.Body>
                <Card.Title>{username}</Card.Title>
                <Card.Text>{blurb}</Card.Text>
            </Card.Body>
        </Card>
    );
};

export default GitHubCard;