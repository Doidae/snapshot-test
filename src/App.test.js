import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
import GitHubCard from './GitHubCard';

test('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders GitHubCard component correctly', () => {
  const { container } = render(
    <GitHubCard
      username="YourUsername"
      imageUrl="path_to_your_image"
      blurb="Your blurb goes here"
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});