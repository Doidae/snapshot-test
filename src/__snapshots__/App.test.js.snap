// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GitHubCard component correctly 1`] = `
<div
  class="card"
  style="width: 18rem;"
>
  <img
    class="card-img-top"
    src="path_to_your_image"
  />
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      YourUsername
    </div>
    <p
      class="card-text"
    >
      Your blurb goes here
    </p>
  </div>
</div>
`;

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
